app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollStateTest.kt

```diff
@@ -208,27 +208,27 @@ class HorizontalSnapScrollStateTest {
                 roomsCount = 5
             )
         ) {
-            assertThat(calculateOnTouchColumnIndex(scrollX = columnWidth * 2)).isEqualTo(0)
+            assertThat(calculateOnTouchColumnIndex(scrollX = columnWidth * 2)).isEqualTo(-1)
         }
     }
 
     @Test
-    fun `calculateOnTouchColumnIndex returns the last possible index as the maximum value in portrait mode to prevent scrolling off the screen to the right`() {
+    fun `calculateOnTouchColumnIndex returns -1 if the target index is higher than the max allowed in portrait mode to prevent scrolling off the screen to the right`() {
         with(
             createState().copy(
                 columnWidth = 993,
                 displayColumnCount = 1,
-                activeColumnIndex = 4,
+                activeColumnIndex = 10,
                 roomsCount = 10
             )
         ) {
             val scrollThresholdWidth = (columnWidth * SCROLL_THRESHOLD_FACTOR).toInt()
-            assertThat(calculateOnTouchColumnIndex(scrollX = -scrollThresholdWidth + 1)).isEqualTo(4)
+            assertThat(calculateOnTouchColumnIndex(scrollX = -scrollThresholdWidth + 1)).isEqualTo(-1)
```

Comment:
Following up on my former comment. I wonder what `activeColumnIndex` is `10`. If I am not mistaken the column index is zero-based which means that the last index in this example would be `9`. With this value the test does not succeed. This again pointed me at the scroll calculation. Shouldn't it be `-scrollThresholdWidth - 1` instead of `-scrollThresholdWidth + 1`?

Reply:
Last Index is indeed 9, the test failed exactly for the reason you mentioned, it should be `-scrollThresholdWidth - 1` :facepalm: